ERCS/erc-7526.md

```diff
@@ -0,0 +1,128 @@
+
+The new token owner MAY make a royalty payment of any amount, which MUST be accepted by the token contract via a dedicated function, `payRoyalty(uint256 tokenId) payable`.
+Upon receipt of royalties, the respective take-back set MUST be emptied (i.e. transfer permissions collapse back to standard ERC-721 rules).
+After each token transfer, there SHOULD be a grace period during which the token is frozen and take-backs are disabled to allow the new owner to decide to either pay the royalty or allow the take-back set to grow.
+If such a grace period is implemented, tokens MUST be unfrozen upon receiving royalties, or at the end of the period.
+
+In addition to clearing the take-back set, receipt of royalties MUST begin a temporary window during which *any* address MAY purchase the token for a price that is a function of the royalty received (this can be viewed as the inverse of the `royaltyInfo()` function of ERC-2981).
+This process is referred to as *auto-listing* and is not limited to addresses in the take-back set.
+The token MUST NOT be transferrable for the duration of the auto-listing.
+Implementations MAY specify the duration of the auto-listing window at their discretion but it MUST be finite and SHOULD be long enough to allow arbitrage opportunities to be detected.
+After expiration of the auto-listing, transfer permissions MUST collapse back to standard ERC-721 rules.
+
+In the event of a sale due to an auto-listing:
+
+1. The token MUST be transferred to the address paying the auto-listing price;
+2. The price MUST be transferred to the original token owner;
+3. The corresponding royalties MUST be reimbursed (i.e. the same address receives both the returned royalties and the sale price); and
+4. The new owner MUST be subject to the same mechanism rules; i.e. take-back, royalty payment, and auto-listing.
+
+If no sale occurs during the auto-listing window, the contract SHOULD transfer the royalty payment to the NFT creator.
+
+Marketplaces facilitating sales SHOULD forward royalties in the same transaction as the sale to guarantee payment within the grace period, thus immediately unfreezing the token.
+Marketplaces MAY allow buyers to configure either the auto-listing price or the royalties paid (as the outcome is equivalent), or they MAY provide sensible default values or employ more sophisticated heuristics at their discretion.
+If allowing buyers to configure values, marketplaces MAY limit this to the auto-listing price for ease of understanding as royalties are then conceptually equivalent to ERC-2981's `royaltyInfo(autoListingPrice)`, which is a concept familiar to end users.
+
+```solidity
+/**
+ * @dev This constitutes a rough draft of the minimum required functions, and will almost certainly change as we
+ * complete the reference implementation.
+ */
```

Comment:
Proposals should be written as if they are already complete and ready to use.

If you want to leave yourself notes, you can use HTML-style comments (`<!-- TODO -->`) though they'll have to be outside of the code fence.